compiler/pipes/convert-spread-operator.cpp

```diff
@@ -0,0 +1,82 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2021 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "compiler/pipes/convert-spread-operator.h"
+
+#include "compiler/compiler-core.h"
+#include "compiler/inferring/public.h"
+
+VertexAdaptor<op_array> array_vertex_from_slice(const VertexConstRange &args, size_t start, size_t end) {
+  return VertexAdaptor<op_array>::create(
+    std::vector<VertexPtr>{args.begin() + start, args.begin() + end}
+  );
+}
+
+VertexPtr ConvertSpreadOperatorPass::on_exit_vertex(VertexPtr root) {
```

Comment:
common comment:
Creating array for non array elements is not a simple operation and it works slow. 
I believe, that we can do better and avoid of creating useless arrays.